src/Payum/Paypal/ExpressCheckout/Nvp/Action/CancelAction.php

```diff
@@ -41,9 +37,12 @@ public function execute($request)
      */
     public function supports($request)
     {
-        return
-            $request instanceof Cancel &&
-            $request->getModel() instanceof \ArrayAccess
-        ;
+        if (false == ($request instanceof Cancel && $request->getModel() instanceof \ArrayAccess)) {
+            return false;
+        }
+
+        // it must take into account only common payments, recurring payments must be cancelled by another action.
+        $model = $request->getModel();
+        return isset($model['TRANSACTIONID']);
```

Comment:
I agree it's BC break, but I was wondering why would you send cancel request with empty model. Will fix.

However, isn't it better to check for empty PROFILEID ?